### ISSUE TITLE
Cannot return slices from an if expression

### ISSUE THREAD

**Issue by 0xrishabh** [noir-lang/noir#2536]
### Aim

```
global DEPTH: Field = 11;
fn main(x: [u8; DEPTH], y: u8) {
    clear(x, y);
}
fn clear<N>(x: [u8; DEPTH], y: u8) -> [u8]{ 
    let mut new_x: [u8] = [0];
    for i in 0..x.len(){
        let b: u8 = i as u8;
        if b >= y {
            new_x[i] = x[i];
        }
        
    }
    new_x
}
```

### Expected Behavior

Should have populated the array?

### Bug

<img width="545" alt="image" src="https://github.com/noir-lang/noir/assets/38362547/ef411f2b-1c4b-4b75-9c7f-60397b51fe01">

### Installation Method

None

### Would you like to submit a PR for this Issue?

Yes

**Comment by vezenovm** [comment]
Could you post which version of Noir you are using? This panic should no longer be hit as it was removed in this PR (https://github.com/noir-lang/noir/pull/2347) which is included in release 0.10.4.

**Comment by 0xrishabh** [comment]
Yeah, you are right the error is not there in 0.10.4, but I ran the same code with the new version and it gives the error below now. @vezenovm 
<img width="548" alt="image" src="https://github.com/noir-lang/noir/assets/38362547/470692bd-bf3f-4728-a178-59a1630696bc">

**Comment by vezenovm** [comment]
This panic is not descriptive enough, but this should be fixed in this PR (https://github.com/noir-lang/noir/pull/2446). I will run this example against it to check. The PR is currently blocked by another bug that I am working on.

**Comment by vezenovm** [comment]
Looks like this issue is resolved with PR #2446, this issue will be updated once that PR is merged.
